File: app/src/main/java/sk/romanstrazanec/particleflowkotlin/javacode/UpdateThread.java
package sk.romanstrazanec.particleflowkotlin.javacode;

import android.os.Handler;

class UpdateThread extends Thread {

    private final Handler updatingHandler;

    UpdateThread(Handler updatingHandler) {
        this.updatingHandler = updatingHandler;
    }

    @Override
    public void run() {
        while (true) {
            try {
                sleep(50);
            } catch (Exception ignored) {
            }
            if (updatingHandler != null) updatingHandler.sendEmptyMessage(0);
        }
    }
}

File: app/src/main/java/sk/romanstrazanec/particleflowkotlin/UpdateThread.kt
package sk.romanstrazanec.particleflowkotlin

import android.os.Handler

class UpdateThread(private var updatingHandler: Handler) : Thread() {
    override fun run() {
        while (true) {
            try {
                Thread.sleep(50)
            } catch (e: Exception) {}
            updatingHandler.sendEmptyMessage(0)
        }
    }
}